src/Sylius/Bundle/CoreBundle/DependencyInjection/SyliusCoreExtension.php

```diff
@@ -179,6 +180,47 @@ private function prependSyliusOrderBundle(ContainerBuilder $container, array $co
         ]);
     }
 
+    private function prependDoctrineExtensionsBundle(ContainerBuilder $containerBuilder): void
+    {
+        if (!$containerBuilder->hasExtension('stof_doctrine_extensions')) {
+            return;
+        }
+
+        $metadataCacheServiceId = null;
+        $doctrineConfig = $containerBuilder->getExtensionConfig('doctrine');
+
+        foreach ($doctrineConfig as $config) {
+            if (isset($config['orm']['entity_managers']['default']['metadata_cache_driver']['id'])) {
+                $metadataCacheServiceId = $config['orm']['entity_managers']['default']['metadata_cache_driver']['id'];
+                break;
+            }
+        }
+
+        $frameworkConfig = $containerBuilder->getExtensionConfig('framework');
+        /** @var array<string> $availableMetadataCachePools */
+        $availableMetadataCachePools = [];
+
+        foreach ($frameworkConfig as $config) {
+            if (isset($config['cache']['pools'])) {
+                $availableMetadataCachePools = array_merge($availableMetadataCachePools, array_keys($config['cache']['pools']));
+            }
+        }
+
+        $metadataCachePool = null;
+        $metadataCacheServiceDefinition = $containerBuilder->getDefinition($metadataCacheServiceId);
+
+        foreach ($metadataCacheServiceDefinition->getArguments() as $argument) {
+            if (in_array((string) $argument, $availableMetadataCachePools, true)) {
+                $metadataCachePool = (string) $argument;
+                break;
+            }
+        }
+
+        $containerBuilder->prependExtensionConfig('stof_doctrine_extensions', [
+            'metadata_cache_pool' => $metadataCachePool,
```

Comment:
This doesn't override an app-level config if someone sets this in `config/packages/stof_doctrine_extensions.yaml`, does it?

Reply:
Nope, as it prepends the config, so all the other changes should be applied on top of it. At least from what I remember 😅. Anyway, this is a PoC, so there's a high probability it won't look like this :D.